sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RegistrationKeyManager.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import android.util.Log;
+
+import com.azure.android.core.logging.ClientLogger;
+import com.azure.android.core.util.Base64Util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.Stack;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+/**
+ * A singleton wrapper which manage the persistence and rotation of secrete keys and execution result related to #{RenewalTokenWorker}.
+ * The secrete keys are persisted and loaded using key-store API. We generate new secrete keys every refresh and evict expired keys
+ * from key-store every refresh.
+ *
+ * RegistrationKeyStore is used as object to manage the keys in file/memory.
+ */
+public final class RegistrationKeyManager {
+    private static RegistrationKeyManager registrationKeyManager;
+
+    private KeyGenerator keyGenerator;
+
+    private RegistrationKeyStore registrationKeyStore;
+
+    //The duration we persist keys in key-store
+    public static final int EXPIRATION_TIME_MINUTES = 45;
+
+    private ClientLogger clientLogger = new ClientLogger(RegistrationKeyManager.class);
+
+    private RegistrationKeyManager()  {
+        try {
+            this.keyGenerator = KeyGenerator.getInstance("AES");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("KeyGenerator failed: " + e.getMessage());
```

Comment:
Don't throw exceptions directly, log them using `logExceptionAsError()` in `ClientLogger` and then throw the Exception returned by said method.
```suggestion
            throw clientLogger.logExceptionAsError(new RuntimeException("KeyGenerator failed: " + e.getMessage()));
```